src/dotnet/commands/dotnet-run/LaunchSettings/ProjectLaunchSettingsProvider.cs

```diff
@@ -32,11 +86,75 @@ public LaunchSettingsApplyResult TryApplySettings(JObject document, JObject mode
             return new LaunchSettingsApplyResult(true, null, config.LaunchUrl);
         }
 
+        private static bool TryGetBooleanValue(JsonElement element, out bool value)
+        {
+            switch (element.Type)
+            {
+                case JsonValueType.True:
+                    value = true;
+                    return true;
+                case JsonValueType.False:
+                    value = false;
+                    return true;
+                case JsonValueType.Number:
+                    if (element.TryGetDouble(out var doubleValue))
+                    {
+                        value = doubleValue != 0;
+                        return true;
+                    }
+                    value = false;
+                    return false;
+                case JsonValueType.String:
+                    return bool.TryParse(element.GetString(), out value);
+                default:
+                    value = false;
+                    return false;
+            }
+        }
+
+        private static bool TryGetStringValue(JsonElement element, out string value)
```

Comment:
@bartonjs, can you please review this code. Particularly, the semantics of this method vs `element.ToString()`

Reply:
The differences I see:

* JsonValueType.True returns `true` here and `True` (`bool.TrueString`) from ToString
* False here returns `false` vs `False`
* Null here returns `null` vs `""`
* Number here normalizes the input to double.ToString(), vs turning the raw input into a string (e.g. 1e1 becomes "10" vs "1e1")
  * Values beyond double.MaxValue will report `"Infinity"` here on netcoreapp3.0, and `null` on netcoreapp2.1
  * Similar for `"-Infinity"`

The first three should mean that the behavior is now different from JSON.Net, unless there was another stringifier in JSON.Net and they didn't agree.  (JsonElement.ToString() was based on JValue.ToString() for these three)